tests/Fixtures/TestBundle/Entity/FilterWithStateOptionsEntity.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity]
+class FilterWithStateOptionsEntity
+{
+    public function __construct(
+        #[ORM\Column(type: 'integer')]
+        #[ORM\Id]
+        #[ORM\GeneratedValue(strategy: 'AUTO')]
+        public ?int $id = null,
+        #[ORM\Column(type: 'date_immutable', nullable: true)]
+        public ?\DateTimeImmutable $dummyDate = null,
```

Comment:
i would make `dummyDate` on the `FilterWithStateOptionsEntity` and `FilterWithStateOptions` named differently, and still make sure filtering and the `QueryParameter` works

Reply:
This will work only once we've access to https://github.com/symfony/symfony/pull/51741. For now unless you do the mapping yourself, the Resource should have the same properties as the Entity.